packages/legend-studio/src/stores/ExecutionPlanState.ts

```diff
@@ -42,30 +48,41 @@ export class ExecutionPlanState {
     | undefined = undefined;
   sqlSelectedTab: SQL_DISPLAY_TABS = SQL_DISPLAY_TABS.SQL_QUERY;
   viewMode: EXECUTION_PLAN_VIEW_MODE = EXECUTION_PLAN_VIEW_MODE.FORM;
+  plan?: ExecutionPlan | object;
+  isGenerating = false;
 
   constructor(editorStore: EditorStore) {
     makeObservable(this, {
       displayData: observable,
       displayDataJson: observable,
       sqlSelectedTab: observable,
       viewMode: observable,
+      isGenerating: observable,
+      plan: observable,
```

Comment:
what do you think if we separate this and make `plan?: ExecutionPlan` and `rawPlan!: object` and do `observable.ref` for the latter? When we can't build the plan, we can make `plan = undefined` and thus fall back to display JSON?

Reply:
So I would favor having the following:

`
plan?: ExecutionPlan
rawPlan?: object
`
Thus when closing the modal we set both to undefined. When generating the plan we populate rawPlan always and if possible we also pollute the plan. 

Thoughts?

Reply:
oh so I forgot we use `.plan` as dialog trigger. Then yeah I agree with your suggestion there. But we should do something like this, when `rawPlan` is set, we will show the plan viewer, but as we build it into `plan` we should show some loading indicator. If we load successfully, we will show the proper execution plan viewer, otherwise, fallback to the old JSON plan viewer.